web/models/users.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Notox.User do
+  use Notox.Web, :model
+
+  @required_fields [
+    :username
+  ]
+
+  @optional_fields @required_fields
+
+  schema "users" do
+    field :username, :string
+    field :password, :string, virtual: true
+    field :password_hash, :string
+
+    timestamps
+  end
+
+  def changeset(model, params \\ %{}) do
+    model
+    |> cast(params, @optional_fields)
+    |> validate_required(@required_fields)
+  end
+
+  defp put_pass_hash(changeset) do
+    case changeset do
+      %Ecto.Changeset{valid?: true, changes: %{password: password}} ->
+        put_change(changeset, :password_hash, Comeonin.Bcrypt.hashpwsalt(password))
+      _ ->
+        changeset
+    end
+  end
+
+  def registration_changeset(model, params \\ %{}) do
+    model
+    |> changeset(params)
+    |> cast(params, ~w(password), [])
+    |> validate_length(:password, min: 6, max: 100)
+    |> put_pass_hash()
```

Comment:
When I try to create a user with the same username as another user, I see this:

![users 2016-09-05 19-08-15](https://cloud.githubusercontent.com/assets/466493/18258041/271dd46c-739c-11e6-9a6d-35c51ad0354c.png)